zproject/backends.py

```diff
@@ -866,6 +886,45 @@ def sync_custom_profile_fields_from_ldap(
         except SyncUserException as e:
             raise ZulipLDAPException(str(e)) from e
 
+    def sync_groups_from_ldap(self, user_profile: UserProfile, ldap_user: _LDAPUser) -> None:
+        if not settings.LDAP_SYNC_GROUPS:
+            return
+        try:
+            ldap_logger.debug("found groups for user: %s", str(user_profile))
+            ldapGroupNameSet = set(ldap_user.group_names)
+
+            memberGroupNames = [
+                membership.user_group.name
+                for membership in UserGroupMembership.objects.filter(user_profile=user_profile)
+            ]
+
+            ldap_logger.debug("user groups: %s", repr(memberGroupNames))
+
+            for group in ldapGroupNameSet:
+                ldap_logger.debug("group: %s", str(group))
+
+                if group not in memberGroupNames and group in self._allZulipGroupNameMap.keys():
+                    ldap_logger.debug("add %s to %s", user_profile.full_name, str(group))
+                    newGroupMembership = UserGroupMembership.objects.create(
+                        user_group=self._allZulipGroupNameMap[group],
+                        user_profile=user_profile,
+                    )
+                    newGroupMembership.save()
```

Comment:
`.save()` is redundant, `.create(` already makes the change to the database

Reply:
see fixup https://github.com/zulip/zulip/pull/23039/commits/2d34bd531f8eb21c75e09ede47f0d0dd8e5b946a